ui/app/pages/permissions-connect/choose-account/choose-account.component.js

```diff
@@ -76,31 +129,79 @@ export default class ChooseAccount extends Component {
     )
   }
 
-  render () {
-    const { originName, selectNewAccountViaModal, permissionsRequestId, cancelPermissionsRequest } = this.props
+  renderAccountsListHeader () {
     const { t } = this.context
+    const { selectNewAccountViaModal, accounts } = this.props
     return (
-      <div className="permissions-connect-choose-account">
-        <div className="permissions-connect-choose-account__title">
-          { t('chooseAnAcount') }
+      <div className="permissions-connect-choose-account__accounts-list-header">
+        <div className={classnames('permissions-connect-choose-account__select-all', {
+          invisible: accounts.length < 2,
+        })}
+        >
+          <CheckBox
+            className="permissions-connect-choose-account__header-check-box"
+            checked={this.allAreSelected()}
+            onClick={() => (this.allAreSelected() ? this.deselectAll() : this.selectAll())}
+          />
+          <div className="permissions-connect-choose-account__text-grey">{ this.context.t('selectAll') }</div>
+          <Tooltip
+            position="bottom"
+            html={(
+              <div style={{ width: 200, padding: 4 }}>
+                {t('selectingAllWillAllow')}
+              </div>
+            )}
+          >
+            <i className="fa fa-info-circle" />
+          </Tooltip>
         </div>
-        <div className="permissions-connect-choose-account__text">
-          { t('toConnectWith', [originName]) }
+        <div
+          className="permissions-connect-choose-account__text-blue"
+          onClick={() => selectNewAccountViaModal(this.handleAccountClick.bind(this))}
+        >
+          { this.context.t('newAccount') }
         </div>
+      </div>
+    )
+  }
+
+  render () {
+    const {
+      selectAccounts,
+      permissionsRequestId,
+      cancelPermissionsRequest,
+      targetDomainMetadata,
+      accounts,
+    } = this.props
+    const { selectedAccounts } = this.state
+    const { t } = this.context
+    return (
+      <div className="permissions-connect-choose-account">
+        <PermissionsConnectHeader
+          icon={targetDomainMetadata.icon}
+          iconName={targetDomainMetadata.name}
```

Comment:
This should be the domain, not the _name_ of the domain. It's important that we show the origin for security reasons, and the name was intentionally dropped from the design after they found that it wasn't set to a very human-readable/friendly name in most cases in the wild anyway.

Reply:
Fixed in 330b0fe72